### PR TITLE
Worker onboarding

### PR DESCRIPTION
**Pull request by quent043**

### CODE REVIEW

File: src/components/Form/CreateWorkerProfileForm.tsx [hunk 1]
```diff
@@ -73,62 +53,36 @@ function CreateWorkerProfileForm({ callback }: { callback?: () => void }) {
...
+        });
       }
-    } else {
-      openConnectModal();
+
+      if (callback) {
+        callback();
+      }
+
+      setSubmitting(false);
```
Comment (0xRomain) [inline src/components/Form/CreateWorkerProfileForm.tsx]: duplicate

File: src/components/Form/EmailForm.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      const response: any = await verifyEmail(values.email, user.id);
+
+      await refreshWorkerData();
+
+      if (response.status !== 200) {
+        showMongoErrorTransactionToast(response.statusText);
+        return;
+      }
+
+      if (response.status === 200) {
```
Comment (0xRomain) [inline src/components/Form/EmailForm.tsx]: not needed

File: src/components/Form/EmailForm.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      enableReinitialize={true}
+      onSubmit={onSubmit}
+      validationSchema={validationSchema}>
+      {({ isSubmitting }) => (
+        <Form>
+          <div className='border border-info rounded-xl p-6 bg-base-100'>
+            <label className='block'>
+              <span className='text-base-content'>Email: </span>
+              <Field
+                type='text'
```
Comment (0xRomain) [inline src/components/Form/EmailForm.tsx]: use type email

File: src/components/Form/CreateWorkerProfileForm.tsx [hunk 4]
```diff
@@ -142,11 +96,11 @@ function CreateWorkerProfileForm({ callback }: { callback?: () => void }) {
         <Form>
           <div className='grid grid-cols-1 gap-6'>
             <label className='block'>
-              <span className='text-xl font-bold '>title</span>
+              <span className='text-xl font-bold '>email</span>
               <Field
                 type='text'
```
Comment (0xRomain) [inline src/components/Form/CreateWorkerProfileForm.tsx]: use type email

File: src/pages/[domain]/dashboard.tsx [hunk 5]
```diff
@@ -11,17 +11,76 @@ import UserServices from '../../components/UserServices';
...
+    useContext(TalentLayerContext);
   const router = useRouter();
-  const { account, user } = useContext(TalentLayerContext);
   const { isBuilderPlaceOwner, builderPlace } = useContext(BuilderPlaceContext);
-  const isComingFromOnboarding = router.asPath.includes('onboarding');
+  const chainId = useChainId();
+  const talentLayerClientConfig = talentLayerClient?.getChainConfig(chainId);
+  const isComingFromOnboarding = router.asPath.includes('onboarding') && isBuilderPlaceOwner;
+
+  const [show, setShow] = useState(false);
```
Comment (0xRomain) [inline src/pages/[domain]/dashboard.tsx]: like others modal, handle this state inside the component

File: src/pages/[domain]/dashboard.tsx [hunk 6]
```diff
@@ -91,7 +150,31 @@ function Dashboard() {
           )}
           {!isBuilderPlaceOwner && (
             <>
-              <div className='mb-12'>
+              <EmailModal show={show} setShow={setShow} />
+              {!workerData?.emailVerified && (
```
Comment (0xRomain) [inline src/pages/[domain]/dashboard.tsx]: create new component VerifyEmailNotification that will have: 
- callback logic 
- render
- data needed (workerdata..)
- display condition

same ActiveDelegationNotification

File: src/pages/[domain]/worker-onboarding/step2.tsx [hunk 7]
```diff
@@ -1,11 +1,83 @@
...
+import Loading from '../../../components/Loading';
+import OnboardingSteps from '../../../components/OnboardingSteps';
+import CreateWorkerId from '../../../components/Home/CreateWorkerId';
+import Steps from '../../../components/Steps';
+import { useSetWorkerProfileOwner } from '../../../modules/BuilderPlace/hooks/UseSetWorkerProfileOwner';
 
 function workerOnboardingStep2() {
+  const { account, loading, user } = useContext(TalentLayerContext);
+  const { mutateAsync: updateWorkerProfileAsync } = useSetWorkerProfileOwner();
+  const [profileAlreadyHasOwner, setProfileAlreadyHasOwner] = useState(false);
```
Comment (0xRomain) [inline src/pages/[domain]/worker-onboarding/step2.tsx]: add commentary to explain this edge case

File: src/pages/[domain]/worker-onboarding/step2.tsx [hunk 8]
```diff
@@ -1,11 +1,83 @@
...
+    const setOwner = async () => {
+      if (account?.isConnected && user?.id && id) {
+        try {
+          const response = await updateWorkerProfileAsync({
+            id,
+            talentLayerId: user.id,
+          });
+          if (response?.error === 'Profile already has an owner.') {
+            setProfileAlreadyHasOwner(true);
+          } else if (response?.id || response?.error) {
```
Comment (0xRomain) [inline src/pages/[domain]/worker-onboarding/step2.tsx]: - no need to have else if for success case
- need to manage error differently

File: src/modules/BuilderPlace/actions.ts [hunk 9]
```diff
@@ -228,3 +235,191 @@ export const updateDomain = async (builderPlace: UpdateBuilderPlaceDomain) => {
...
+  }
+};
+
+export async function checkUserEmailVerificationStatus(
+  userId: string,
+  res: NextApiResponse,
+): Promise<any> {
+  try {
+    await connection();
+    const worker = await Worker.findOne({ talentLayerId: userId });
```
Comment (0xRomain) [inline src/modules/BuilderPlace/actions.ts]: make the findOne directly in the api and pass worker as params here

File: src/modules/BuilderPlace/actions.ts [hunk 10]
```diff
@@ -228,3 +235,191 @@ export const updateDomain = async (builderPlace: UpdateBuilderPlaceDomain) => {
...
+    }
+    console.log('Email verified');
+    return worker;
+  } catch (error) {
+    res.status(500).json({ error: error });
+  }
+}
+
+export async function checkOrResetTransactionCounter(
+  worker: any,
```
Comment (0xRomain) [inline src/modules/BuilderPlace/actions.ts]: need typing

File: src/modules/BuilderPlace/actions.ts [hunk 11]
```diff
@@ -228,3 +235,191 @@ export const updateDomain = async (builderPlace: UpdateBuilderPlaceDomain) => {
...
+): Promise<any> {
+  try {
+    await connection();
+
+    const nowMilliseconds = new Date().getTime();
+    const oneWeekAgoMilliseconds = new Date(nowMilliseconds - 7 * 24 * 60 * 60 * 1000).getTime(); // 7 days ago
+
+    if (worker.counterStartDate > oneWeekAgoMilliseconds) {
+      // Less than one week since counterStartDate
+      if (worker.weeklyTransactionCounter >= 50) {
```
Comment (0xRomain) [inline src/modules/BuilderPlace/actions.ts]: add 50 as constant

File: src/modules/BuilderPlace/actions.ts [hunk 12]
```diff
@@ -228,3 +235,191 @@ export const updateDomain = async (builderPlace: UpdateBuilderPlaceDomain) => {
...
+      throw new Error('Email not verified');
+    }
+    console.log('Email verified');
+    return worker;
+  } catch (error) {
+    res.status(500).json({ error: error });
+  }
+}
+
+export async function checkOrResetTransactionCounter(
```
Comment (0xRomain) [inline src/modules/BuilderPlace/actions.ts:395]: Create a trello ticket to: 
- make this check on the frontend and automatically disable delegation if limit reach
- display on the user menu the current transaction counter

Comment (quent043) [reply src/modules/BuilderPlace/actions.ts:395]: Done here

**Pull request merged**